### PR TITLE
8316190: Improve MemorySegment::toString

### PR DESCRIPTION
**Pull request by minborg**
This PR proposes to improve the MemorySegment::toString to reduce cluttering and add missing comas.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316190](https://bugs.openjdk.org/browse/JDK-8316190): Improve MemorySegment::toString (**Enhancement** - P4)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15740/head:pull/15740` \
`$ git checkout pull/15740`

Update a local copy of the PR: \
`$ git checkout pull/15740` \
`$ git pull https://git.openjdk.org/jdk.git pull/15740/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15740`

View PR using the GUI difftool: \
`$ git pr show -t 15740`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15740.diff">https://git.openjdk.org/jdk/pull/15740.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15740#issuecomment-1719327761)

### CODE REVIEW

File: test/jdk/java/foreign/TestSegments.java [hunk 1]
```diff
@@ -233,7 +248,7 @@ public Object[][] segmentFactories() {
                 () -> MemorySegment.ofArray(new double[] { 1d, 2d, 3d, 4d} ),
                 () -> MemorySegment.ofArray(new float[] { 1.0f, 2.0f, 3.0f, 4.0f }),
                 () -> MemorySegment.ofArray(new int[] { 1, 2, 3, 4 }),
-                () -> MemorySegment.ofArray(new long[] { 1l, 2l, 3l, 4l } ),
+                () -> MemorySegment.ofArray(new long[] { 1L, 2L, 3L, 4L } ),
```
Comment (minborg) [inline test/jdk/java/foreign/TestSegments.java:251]: Drive-by fix

File: src/java.base/share/classes/jdk/internal/foreign/Utils.java [hunk 2]
```diff
@@ -273,4 +273,9 @@ public static <L extends MemoryLayout> L wrapOverflow(Supplier<L> layoutSupplier
     public static boolean containsNullChars(String s) {
         return s.indexOf('\u0000') >= 0;
     }
+
+    public static String toHexString(long value) {
```
Comment (minborg) [inline src/java.base/share/classes/jdk/internal/foreign/Utils.java:277]: We intend to use this utility method in coming PRs.

Comment (RogerRiggs) [reply src/java.base/share/classes/jdk/internal/foreign/Utils.java:277]: The String concat code is really good at optimizing, but this will require a new string be created.
$0.02,Roger

**Review by RogerRiggs — APPROVED** [review]

File: src/java.base/share/classes/jdk/internal/foreign/Utils.java (hunk 2 above)
Comment (cl4es) [reply src/java.base/share/classes/jdk/internal/foreign/Utils.java:277]: Why not `HexFormat.of().withPrefix("0x").formatHexString(long)` I thought, then realized there's no method such method that takes a `long` so that the "0x" + hex digits can be done in a single step without concatenation. Seems there might be room for improvement here.
